test/streaming-client.spec.ts

```diff
@@ -0,0 +1,188 @@
+import { PassThrough } from 'stream';
+import { WebSocketClient, WebSocketConnection } from 'websocket';
+import { AudioConfig } from '../src/models/streaming/AudioConfig';
+import { BufferedDuplex } from '../src/models/streaming/BufferedDuplex';
+import { RevAiStreamingClient } from '../src/streaming-client';
+const fs = require('fs');
+const events = require('events');
+
+jest.useFakeTimers();
+
+let sut: RevAiStreamingClient;
+let mockClient: WebSocketClient;
+
+const audioConfig = new AudioConfig("audio/x-wav");
+const token = "testToken";
+
+describe('streaming-client', () => {
+    beforeEach(() => {
+        sut = new RevAiStreamingClient(token, audioConfig);
+        mockClient = new WebSocketClient();
+        sut.client = mockClient;
+    });
+
+    describe('start', () => {
+        it('Connects to api with parameters', () => {
+            const res = sut.start();
+
+            expect(mockClient.connect).toBeCalledWith(`wss://api.rev.ai/speechtotext/v1alpha/stream?` +
+                `access_token=${token}` +
+                `&content_type=${audioConfig.getContentTypeString()}`
+            );
+            expect(mockClient.connect).toBeCalledTimes(1);
+        });
+
+        it('Returns duplex stream', () => {
+            const res = sut.start();
+
+            expect(res).toBeInstanceOf(BufferedDuplex);
+        });
+    });
+
+    describe('client setup', () => {
```

Comment:
I think client setup tests should be a part of `start` tests.

We shouldn't have tests for private functions.

Reply:
You can rephrase the test names "addsEventListener to httpResponse", etc.